Demo/Demo/SwiftUIComponents/PayPalVaultViews/PayPalVaultView.swift

```diff
@@ -14,10 +14,11 @@ struct PayPalVaultView: View {
                         paymentSourceType: PaymentSourceType.paypal(usageType: "MERCHANT")
                     )
                     SetupTokenResultView(vaultViewModel: paypalVaultViewModel)
-                    if let url = paypalVaultViewModel.state.setupToken?.paypalURL {
+                    if let urlString = paypalVaultViewModel.state.setupToken?.paypalURL,
+                        let setupTokenID = paypalVaultViewModel.state.setupToken?.id {
                         Button("Vault PayPal") {
                             Task {
-                                await paypalVaultViewModel.vault(url: url)
+                                await paypalVaultViewModel.vault(url: urlString, setupTokenID: setupTokenID)
```

Comment:
This may be a silly question but is there a reason our view is pulling these values from the view model only to send it to the view model? Can we just call `await paypalVaultViewModel.vault()` with nothing then pull the setup token and id from the state within the view model's vault method?

Reply:
It's just the way we have the demo app setup. I get the setup token from create API and update the state in viewModel. 

And then to populate next view I get these values that were just created and use the vault function in viewModel to launch the approval url. 

I think it'd be weird for vault function in viewModel to just fetch these values without context.

Reply:
Yeah I guess to me it just seems like a pretty big anti pattern to send data from ViewModel > View > ViewModel since that data already exists in the view model itself. I'm not sure if using something like a `bool` in the view model would be a better way of indicating that we can vault and to show this view. Just worth thinking about if you have ideas.

Reply:
the property is an optional field until it is populated. 
It's a pattern I've used in views before where I choose to present a UI element
depending on presence of some value. 

So rather than checking in the view model's vault function for presence of its values, I just send the values from the view where it checks the value anyway.

Reply:
That we need to launch this url as opposed to mobile SDK directly calling API was researched by Steven
@sshropshire

Reply:
This was guidance from the JS PPCP SDK. Apparently the URL given to us by PPaaS is lower latency than any URL we can create on our own using a setup token ID.

Reply:
Gotcha, my question wasn't around the URL at all. It was around passing view model data from the view model back to the view model, which is a bit of an odd pattern. I know we do it in many places, so should think about if there is a better way vs having the view model passed it's own data. Really just something to think about!

Reply:
Oh I think I answered that part in the wrong section. That portion of my answer was in Sammy's question on the bottom.

Reply:
Yeah, expanding on your point Jax, I don't need to pass it, I just check existence of those values in the view, so I just passed them so I don't have to check for nil values in the function in the viewModel.